src/frontend/src/views/Upload/components/Metadata/index.tsx

```diff
@@ -47,17 +90,46 @@ export default function Metadata({
   const [hasImportedFile, setHasImportedFile] = useState(false);
   const [autocorrectWarnings, setAutocorrectWarnings] =
     useState<SampleIdToWarningMessages>(EMPTY_OBJECT);
+  const [locations, setLocations] = useState<NamedGisaidLocation[]>([]);
 
-  function handleMetadata(result: ParseResult) {
-    const { data: sampleIdToMetadata, warningMessages } = result;
+  const loadLocations = async () => {
+    const result: LocationsResponse = await getLocations();
+    const namedLocations: NamedGisaidLocation[] = result.locations.map(
+      (location) => {
+        return {
+          name: stringifyLocation(location),
+          ...location,
+        };
+      }
+    );
+    setLocations(namedLocations);
+  };
 
+  useEffect(() => {
+    loadLocations();
+  }, [samples]);
```

Comment:
why is this rerun whenever samples change?

Reply:
Yeah, it feels like we probably could just run this once every time the user sees the Metadata component. I think an empty array so it only ever gets done once per visit to the Metadata section would be totally reasonable. That said, once the user gets to Metadata, `samples` should effectively be static -- I'm like 95% sure it never changes after upload samples step 1 -- so this implicitly does that, but could be clearer that's the intention.

Reply:
Fixed!